diff --git a/node_modules/react-native/Libraries/LinkingIOS/RCTLinkingManager.h b/node_modules/react-native/Libraries/LinkingIOS/RCTLinkingManager.h
index 21733a7..28c94c6 100644
--- a/node_modules/react-native/Libraries/LinkingIOS/RCTLinkingManager.h
+++ b/node_modules/react-native/Libraries/LinkingIOS/RCTLinkingManager.h
@@ -10,6 +10,7 @@
 #import <UIKit/UIUserActivity.h>
 #endif
 #import <React/RCTEventEmitter.h>
+#import <UMReactNativeAdapter/UMReactNativeEventEmitter.h>
 
 @interface RCTLinkingManager : RCTEventEmitter
 
